### PR TITLE
CORTX-34324: Add Support for Dashboard application container image generation

### PR DESCRIPTION
**Pull request by shailesh-vaidya**
# Problem Statement
- Added Support for Dashboard application container image generation. 
     - Added `build.sh` and `docker-compose.yml` to support `codacy` and `portscanner` service container images.
     - Added Jenkinsfile and created https://eos-jenkins.colo.seagate.com/job/Dashboard/job/build-dashboard-images/ to generate images.

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM
- [x] Jenkins pipelines used for testing - https://eos-jenkins.colo.seagate.com/job/Dashboard/job/build-dashboard-images/18/

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] Jira ID/ COMMUNITY is prefixed in PR title
      Prefix Syntax - 
        `CORTX-<5 digit JIRA ID>: <PR Title>` - for Seagate Employees
        `COMMUNITY: <PR Title>`  - for open source contributors
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered solutions/dashboard/ElasticsearchKibanaInstallation.md](https://github.com/shailesh-vaidya/cortx-re/blob/dashboard-image-generation/solutions/dashboard/ElasticsearchKibanaInstallation.md)
[View rendered solutions/dashboard/README.md](https://github.com/shailesh-vaidya/cortx-re/blob/dashboard-image-generation/solutions/dashboard/README.md)
[View rendered solutions/dashboard/deployment/README.md](https://github.com/shailesh-vaidya/cortx-re/blob/dashboard-image-generation/solutions/dashboard/deployment/README.md)

### CODE REVIEW

**Review by Sanket292001 — APPROVED** [review]

File: solutions/dashboard/build/build.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+set -e -o pipefail
+
+usage() {
+
+echo "Generate CORTX container images from provided CORTX release build"
```
Comment (gauravchaudhari02) [inline solutions/dashboard/build/build.sh:25]: we are generating dashboard images right?

File: solutions/dashboard/build/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
```
Comment (gauravchaudhari02) [inline solutions/dashboard/build/docker-compose.yml:2]: please change year to 2022

File: solutions/dashboard/deployment/dashboard/codacy.yaml [hunk 3]
```diff
@@ -13,7 +13,7 @@ spec:
         spec:
           containers:
             - name: dashboard-codacy
-              image: docker.io/sanket02092001/scs:1.0.0
+              image: cortx-docker.colo.seagate.com/dashboard/codacy:1.0.0
```
Comment (gauravchaudhari02) [inline solutions/dashboard/deployment/dashboard/codacy.yaml:16]: what if someone uploaded his/her images on local registry?

File: solutions/dashboard/build/build.sh (hunk 1 above)
Comment (shailesh-vaidya) [reply solutions/dashboard/build/build.sh]: Fixed. Please review again

File: solutions/dashboard/build/docker-compose.yml (hunk 2 above)
Comment (shailesh-vaidya) [reply solutions/dashboard/build/docker-compose.yml]: Fixed. Please review it again.

File: solutions/dashboard/deployment/dashboard/codacy.yaml (hunk 3 above)
Comment (shailesh-vaidya) [reply solutions/dashboard/deployment/dashboard/codacy.yaml]: This use case needs to be explored. Maybe we need to create a helm chart and create value.yaml that can be updated with custom image details.  Developer needs to update YAML files directly as of now.

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**